zivid_camera/src/zivid_camera.cpp

```diff
@@ -379,6 +366,72 @@ bool ZividCamera::capture2DServiceHandler(Capture::Request&, Capture::Response&)
+      std::chrono::round<std::chrono::milliseconds>(std::chrono::duration<double>{ req.max_capture_time.toSec() });
+  const auto ambient_light_frequency = [&req]() {
+    switch (req.ambient_light_frequency)
+    {
+      case CaptureAssistantSuggestSettings::Request::AMBIENT_LIGHT_FREQUENCY_NONE:
+        return Zivid::CaptureAssistant::AmbientLightFrequency::none;
+      case CaptureAssistantSuggestSettings::Request::AMBIENT_LIGHT_FREQUENCY_50HZ:
+        return Zivid::CaptureAssistant::AmbientLightFrequency::hz50;
+      case CaptureAssistantSuggestSettings::Request::AMBIENT_LIGHT_FREQUENCY_60HZ:
+        return Zivid::CaptureAssistant::AmbientLightFrequency::hz60;
+    }
+    throw std::runtime_error("Unhandled AMBIENT_LIGHT_FREQUENCY value: " + std::to_string(req.ambient_light_frequency));
+  }();
+
+  Zivid::CaptureAssistant::SuggestSettingsParameters suggest_settings_parameters(max_capture_time,
+                                                                                 ambient_light_frequency);
+  ROS_INFO_STREAM("Getting suggested settings using parameters: " << suggest_settings_parameters);
+  const auto suggested_settings{ Zivid::CaptureAssistant::suggestSettings(camera_, suggest_settings_parameters) };
+
+  if (suggested_settings.empty())
+  {
+    throw std::runtime_error("The suggestSettings function returned 0 settings!");
+  }
+  if (suggested_settings.size() > capture_frame_config_dr_servers_.size())
+  {
+    throw std::runtime_error("The number of suggested settings (" + std::to_string(suggested_settings.size()) + ") " +
+                             "is larger than the number of dynamic_reconfigure capture/frame_<n> servers (" +
+                             std::to_string(capture_frame_config_dr_servers_.size()) + "). Increase launch parameter " +
+                             "num_capture_frames. See README.md for more information.");
+  }
+
+  ROS_INFO_STREAM("CaptureAssistant::suggestSettings returned " << suggested_settings.size() << " settings");
+
+  capture_general_config_dr_server_->setConfig(zividSettingsToConfig<CaptureGeneralConfig>(suggested_settings[0]));
+
+  for (std::size_t i = 0; i < suggested_settings.size(); i++)
+  {
+    ROS_DEBUG_STREAM("Updating setting " << i << " to " << suggested_settings[i]);
+    auto config = zividSettingsToConfig<CaptureFrameConfig>(suggested_settings[i]);
+    config.enabled = true;
+    capture_frame_config_dr_servers_[i]->setConfig(config);
+  }
+
+  // Any other frames that are enabled must be disabled
+  for (std::size_t i = suggested_settings.size(); i < capture_frame_config_dr_servers_.size(); i++)
+  {
+    if (capture_frame_config_dr_servers_[i]->config().enabled)
+    {
+      ROS_INFO_STREAM("Frame config " << i << " was enabled, so disabling it");
+      auto config = capture_frame_config_dr_servers_[i]->config();
+      config.enabled = false;
+      capture_frame_config_dr_servers_[i]->setConfig(config);
+    }
+  }
+
+  return true;
```

Comment:
There is no case where this can become false? 
Is there some scenario that it could make sense for the service call to return false, or do you think all these cases should rather be covered by throwing exceptions (as above)?

Reply:
I think the service should either throw or return true. I don't see a good reason for it to return false.